File: Algorithm/Python/674. Longest Continuous Increasing Subsequence.py
# -*- coding: utf-8 -*-
# @Author: WuLC
# @Date:   2017-09-13 19:57:19
# @Last Modified by:   WuLC
# @Last Modified time: 2017-09-13 19:58:09

# be careful of the last element
class Solution(object):
    def findLengthOfLCIS(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) == 0:
            return 0
        result = 1
        count = 1
        for i in xrange(1, len(nums)):
            if nums[i] > nums[i-1]:
                count += 1
            else:
                count = 1
            result = max(result, count)
        return result
                    
                
                

File: Algorithm/C++/674.longest-continuous-increasing-subsequence.cpp
/*
 * @lc app=leetcode id=674 lang=cpp
 *
 * [674] Longest Continuous Increasing Subsequence
 */

// @lc code=start

#include <vector>

class Solution {
public:
    int findLengthOfLCIS(std::vector<int>& nums) {
        int curr_len = 1, result = 1;
        for (int i = 1; i < nums.size(); i++) {
            if (nums[i] > nums[i-1]) {
                curr_len++;
            } else {
                curr_len = 1;
            }
            result = std::max(result, curr_len);
        }
        return result;
    }
};
// @lc code=end

